feat(Base Components) Hero Block. DCH-35

queenOfPentacls opened the pull request:
<!-- gfx pull request template -->
<!-- note: do note remove comments -->

<!-- tickets -->
## Ticket(s)

[DCH-35](https://graveflex.atlassian.net/jira/software/projects/DCH/issues/DCH-35)

<!-- /tickets -->

<!-- links -->
## Links

[Testing Page]()
[CMS]()
[Storybook]()

<!-- /links -->

<!-- description -->
## Description

Updated Hero Block, supporting the variants for:

* Image on left
* Image on right
* Image Full-Bleed
* Image Background
* Text/content alignment at any position

<!-- /description -->
<!-- Describe the ask/feature associated with the ticket and how it was implemented -->
<!-- Are there any known/expected issues w/ this update that should be noted for QA? -->

<!-- reproduction -->
## Reproduction Steps

<!-- Provide a brief set of steps to verify/view the update. -->
<!-- /reproduction -->

<!-- checks -->
<!-- /checks -->

queenOfPentacls commented on packages/ui/components/primitives/ResponsivePayloadImage/index.tsx:
```diff
@@ -165,7 +174,8 @@ function ResponsivePayloadImage({
         }
         sizes={fill ? fillSizes : '100vw'}
         alt={alt ?? ''}
-        style={{ width: fill ? '100%' : 'initial' }}
```
Moving these out of inline styles so they can be overrided without extreme css specificity

queenOfPentacls commented on apps/web/src/blocks/HeroBlock/index.tsx:
```diff
@@ -128,28 +153,51 @@ function HeroBlock({
+              <FormWrapper onSubmit={(data) => console.log(data)} cta={cta}>
+                <TextInput
+                  {...textinput}
+                  name="TextInput"
```
Hardcoding label and placeholder because the typing for the `hook-fields` is incompatible with the payload types & needs to be updated

queenOfPentacls commented on apps/web/src/payload/fields/TextInput.ts:
```diff
@@ -0,0 +1,36 @@
+        required: false
+      },
+      {
+        name: 'label',
```
The typing for these fields needs to be updated in `refract`. The label isn't optional, and the typing in refract is incompatible with payload.

siobahnschuck replied:
@queenOfPentacls can you do a write up of what needs to be updated in the refract repo and fwd it to Lynn. This should be resolved before we merge so that the CMS is functional here.

queenOfPentacls commented on apps/web/src/blocks/HeroBlock/index.tsx:
```diff
@@ -128,28 +150,52 @@ function HeroBlock({
+            >
+              <TextInput
+                {...textinput}
+                name="TextInput"
```
Hardcoding for these values because the typing for these fields needs to be updated in refract. The label and placeholder are incompatible with the typing for payload text strings.

siobahnschuck commented on apps/web/src/blocks/HeroBlock/HeroBlock.config.ts:
```diff
@@ -53,6 +25,49 @@ const HeroBlock: Block = {
+    },
     {
       type: 'text',
       name: 'title',
```
Let's update this so that the title and subtitle are handled in the **same** richText `content` field

siobahnschuck reviewed (changes requested):
This looks good! A few small revisions: 
- consolidate `title` and `subtitle` into one richText `content` field
- dynamically show the input field based on content entry (can this also be collapsible in the CMS admin?)
- sync with Lynn to update the input types in refract so that we can pull `label` and `placeholder` text from the CMS.

queenOfPentacls commented:
> * consolidate `title` and `subtitle` into one richText `content` field
> * dynamically show the input field based on content entry (can this also be collapsible in the CMS admin?)
> * sync with Lynn to update the input types in refract so that we can pull `label` and `placeholder` text from the CMS.

* Done
* Added a collapsible to the input fields and updated the form to dynamically show based on placeholder. I think placeholder makes sense to use as the conditional instead of label here because the template designs never feature a floating label.
* I think the typing in refract is fine, actually. Adding the `undefined` fallback fixes the problems I was having w it.
